### PR TITLE
Refator the CreateFlags command

### PR DESCRIPTION
**Pull request by mslacken**
* added the additional type flag, which can be bool,IP
* UNDEF is recognized for the bool flag
* 0.0.0.0 must be used as UNDEF for IP flag

Signed-off-by: Christian Goll <cgoll@suse.de>

### CODE REVIEW

**Comment by mslacken** [comment]
@anderbubble 
This PR brings bag the `--discoverable` flag with any options. 
Any boolean flag can be set this way and will always set flag to true, as warewulf parses unset values to false.
Also IP addresses must be set in the right format.

This maybe a candidate for a 4.4.1 release as the 4.4.0 broke the old commandline behavior which is so restored.

**Comment by anderbubble** [comment]
I'll check it out!

**Comment by anderbubble** [comment]
@mslacken I've been looking at this code, trying to make sure I understand it, and seeing if I could suggest a change that would restore the ability to specify UNSET/UNDEF for mac and IP addresses.

But went I went to just test it at the command-line, I found this, which I think represents an error condition:

```console
$ sudo wwctl node set --discoverable asdf1
ERROR: invalid MAC address
```

`asdf1` doesn't have a MAC address specified.

```yaml
  asdf1:
    profiles:
    - default
    network devices:
      primary:
        device: eno3
```

I wonder if you might be willing to include my refactoring in this PR? https://github.com/anderbubble/warewulf/tree/RefactorFlags

File: internal/pkg/node/methods.go [hunk 1]
```diff
@@ -10,6 +10,10 @@ import (
 	"github.com/hpcng/warewulf/internal/pkg/wwlog"
 )
 
+func GetUnsetVerbs() []string {
+	return []string{"UNSET", "DELETE", "UNDEF", "undef", "--", "nil", "0.0.0.0"}
```
Comment (anderbubble) [inline internal/pkg/node/methods.go:14]: What would be the correct value to use to unset a mac address?

**Review by anderbubble — CHANGES_REQUESTED** [review]

**Comment by mslacken** [comment]
> @mslacken I've been looking at this code, trying to make sure I understand it, and seeing if I could suggest a change that would restore the ability to specify UNSET/UNDEF for mac and IP addresses.
> 
> But went I went to just test it at the command-line, I found this, which I think represents an error condition:
> 
> ```
> $ sudo wwctl node set --discoverable asdf1
> ERROR: invalid MAC address
> ```
> 
$ should be wwctl node set --discoverable --netname asdf1

And yes birngen back UNDEF for IP addresses is better

**Comment by anderbubble** [comment]
@mslacken I merged this locally again tonight, and found this bug, at least:

```console

$ sudo wwctl node add asdf1
ERROR: invalid MAC address
```

**Comment by mslacken** [comment]
@anderbubble should be fixed right now.

**Comment by anderbubble** [comment]
@mslacken looks like `--discoverable=false` still evaluates to true.

```console
# wwctl node set --discoverable=false n1
Are you sure you want to modify 1 nodes(s): y

# wwctl node list -a n1 | grep discoverable
n1                   discoverable       --           true
```

File: internal/pkg/node/datastructure.go [hunk 2]
```diff
@@ -70,17 +70,17 @@ type KernelConf struct {
...
-	OnBoot  string            `yaml:"onboot,omitempty" lopt:"onboot" comment:"Enable/disable network device (yes/no)"`
+	OnBoot  string            `yaml:"onboot,omitempty" lopt:"onboot" comment:"Enable/disable network device (true/false)" type:"bool"`
 	Device  string            `yaml:"device,omitempty" lopt:"netdev" sopt:"N" comment:"Set the device for given network"`
-	Hwaddr  string            `yaml:"hwaddr,omitempty" lopt:"hwaddr" sopt:"H" comment:"Set the device's HW address for given network"`
-	Ipaddr  string            `yaml:"ipaddr,omitempty" comment:"IPv4 address in given network" sopt:"I" lopt:"ipaddr"`
+	Hwaddr  string            `yaml:"hwaddr,omitempty" lopt:"hwaddr" sopt:"H" comment:"Set the device's HW address for given network" type:"MAC"`
+	Ipaddr  string            `yaml:"ipaddr,omitempty" comment:"IPv4 address in given network" sopt:"I" lopt:"ipaddr" type:"IP"`
 	IpCIDR  string            `yaml:"ipcidr,omitempty"`
-	Ipaddr6 string            `yaml:"ip6addr,omitempty" lopt:"ipaddr6" comment:"IPv6 address"`
+	Ipaddr6 string            `yaml:"ip6addr,omitempty" lopt:"ipaddr6" comment:"IPv6 address" type:"IP"`
```
[the remark this answers is not in the document]
Comment (mslacken) [reply internal/pkg/node/datastructure.go:79]: It not its own tye in go, so ParseIP works for ipv4 and ipv6

File: internal/pkg/node/methods.go (hunk 1 above)
Comment (mslacken) [reply internal/pkg/node/methods.go:14]: UNDEF works also for mac addresses

**Comment by mslacken** [comment]
Fixed the parsing of bool, was a realy stupid typo

**Comment by anderbubble** [comment]
@mslacken I merged #721, and now this branch is conflicting with development. Could you rebase?

**Comment by JasonYangShadow** [comment]
LGTM.

**Review by JasonYangShadow — APPROVED** [review]

**Pull request merged**